platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/utility/TemporaryFileBuilder.java

```diff
@@ -44,9 +44,10 @@ private TemporaryFileBuilder() {}
      */
     public static synchronized Path getTemporaryFileLocation() throws IOException {
         if (temporaryFileLocation == null) {
-            final TemporaryFileConfig config = ConfigurationHolder.getConfigData(TemporaryFileConfig.class);
+            final FileSystemManagerConfig config = ConfigurationHolder.getConfigData(FileSystemManagerConfig.class);
             final StateCommonConfig stateConfig = ConfigurationHolder.getConfigData(StateCommonConfig.class);
-            overrideTemporaryFileLocation(getAbsolutePath(config.getTemporaryFilePath(stateConfig)));
+            overrideTemporaryFileLocation(getAbsolutePath(
+                    stateConfig.savedStateDirectory().resolve(config.rootPath()).toString()));
         }
```

Comment:
I suggest we leave this class as is, rename it to `LegacyTemporaryFileBuilder`, and mark it as deprecated. Then we should put pressure on the remaining users of the static version to convert to non-static access patterns.

Reply:
Absolutely. I hold that change because of this [pr](https://github.com/hashgraph/hedera-services/pull/12734/files), but I can also include it here.